src/Components/Poems/Poems.js

```diff
@@ -8,37 +8,28 @@ function Poem() {
   const [selectedAuthor, setSelectedAuthor] = useState('');
   const [selectedAuthorPoems , setSelectedAuthorPoems] = useState([]);
   // not using this state yet.
-  //const [allPoems, setAllPoems] = useState([]);
   const [errMessage, setErrMessage] = useState('');
 
 
   const addToFaves = (e) => {
     let localData = localStorage.getItem('favePoems')
     
-    if(localData) {
-      localData = JSON.parse(localData)
-      //console.log("THIS IS LOCAL DATA inside add2Faves", localData)
-    }
-    
     let matchedPoem = selectedAuthorPoems.filter((poem, index) => index === parseInt(e.target.id))
     
-    //matchedPoem = matchedPoem[0]
-
-    //console.log("This is the matched poem", matchedPoem);
-    //let stringifiedPoem = JSON.stringify(matchedPoem[0])
-
     if(localData) {
-      //localData = JSON.parse(localData)
-      //console.log("LOCAL DATA inside Assignment IF", localData)
+      localData = JSON.parse(localData)
+      let found = localData.find(poem => poem.title === matchedPoem[0].title)
```

Comment:
When searching for a poem & then favoriting songs, this error message appears.
<img width="1062" alt="Screen Shot 2021-08-08 at 3 31 42 PM" src="https://user-images.githubusercontent.com/62954974/128646369-90bfd1bb-b85e-4cc3-a5a8-00a7cb42b287.png">

Reply:
additionally (after getting results from one search result & then searching for another), the previous results do not get removed & the initially searched name is changed to the newly searched name in each of the poem cards

Reply:
Thank you for breaking my app! I will look into the cause of it. And try async & await calls

Reply:
anytime brochacho 😄